File: checkers.php
<?php
// checkers.php
// Версия на PHP 8 с классами, атрибутами (не используются), цветной вывод, JSON сохранение

declare(strict_types=1);

// ANSI
function colorize(string $text, string $fg = "\033[0m", string $bg = ""): string {
    return $fg . $bg . $text . "\033[0m";
}

const RESET = "\033[0m";
const BLACK = "\033[30m";
const RED = "\033[31m";
const GREEN = "\033[32m";
const YELLOW = "\033[33m";
const BLUE = "\033[34m";
const MAGENTA = "\033[35m";
const CYAN = "\033[36m";
const WHITE = "\033[37m";
const BG_BLACK = "\033[40m";
const BG_WHITE = "\033[47m";

enum ColorType: int { case WHITE = 1; case BLACK = 2; }
enum PieceType: int { case MAN = 1; case KING = 2; }

class Piece {
    public function __construct(public ColorType $color, public PieceType $type) {}
}

class Move {
    public function __construct(public int $fromRow, public int $fromCol, public int $toRow, public int $toCol, public array $captures) {}
}

class Board {
    const SIZE = 10;
    public array $grid = [];

    public function __construct() {
        for ($r=0; $r<self::SIZE; $r++) {
            for ($c=0; $c<self::SIZE; $c++) {
                if (($r+$c)%2==1) {
                    if ($r<4) $this->grid[$r][$c] = new Piece(ColorType::WHITE, PieceType::MAN);
                    elseif ($r>5) $this->grid[$r][$c] = new Piece(ColorType::BLACK, PieceType::MAN);
                }
            }
        }
    }

    public function inBounds(int $r, int $c): bool { return $r>=0 && $r<self::SIZE && $c>=0 && $c<self::SIZE; }
    public function getPiece(int $r, int $c): ?Piece { return $this->inBounds($r,$c) ? ($this->grid[$r][$c] ?? null) : null; }
    public function isOccupied(int $r, int $c): bool { return $this->getPiece($r,$c) !== null; }
    public function placePiece(int $r, int $c, Piece $p): void { $this->grid[$r][$c] = $p; }
    public function removePiece(int $r, int $c): void { unset($this->grid[$r][$c]); }
    public function findPieces(ColorType $color): array {
        $res = [];
        for ($r=0; $r<self::SIZE; $r++) for ($c=0; $c<self::SIZE; $c++) {
            if (isset($this->grid[$r][$c]) && $this->grid[$r][$c]->color === $color) $res[] = [$r,$c];
        }
        return $res;
    }
    public function isKingRow(int $r, ColorType $color): bool {
        return ($color === ColorType::WHITE && $r === self::SIZE-1) || ($color === ColorType::BLACK && $r === 0);
    }
    public function promoteToKing(int $r, int $c): void {
        if (isset($this->grid[$r][$c]) && $this->grid[$r][$c]->type === PieceType::MAN) {
            $this->grid[$r][$c] = new Piece($this->grid[$r][$c]->color, PieceType::KING);
        }
    }
    public function clone(): Board {
        $b = new Board();
        for ($r=0; $r<self::SIZE; $r++) for ($c=0; $c<self::SIZE; $c++) {
            if (isset($this->grid[$r][$c])) $b->grid[$r][$c] = new Piece($this->grid[$r][$c]->color, $this->grid[$r][$c]->type);
        }
        return $b;
    }
}

class Game {
    public Board $board;
    public ColorType $currentTurn;
    public bool $twoPlayers;
    public array $moveHistory = [];

    public function __construct(bool $twoPlayers = false) {
        $this->board = new Board();
        $this->currentTurn = ColorType::WHITE;
        $this->twoPlayers = $twoPlayers;
    }

    public function opponent(ColorType $c): ColorType {
        return $c === ColorType::WHITE ? ColorType::BLACK : ColorType::WHITE;
    }

    private function getManMoves(int $row, int $col, ColorType $color): array {
        $moves = [];
        $dirs = $color === ColorType::WHITE ? [[1,-1],[1,1]] : [[-1,-1],[-1,1]];
        foreach ($dirs as [$dr,$dc]) {
            $nr = $row+$dr; $nc = $col+$dc;
            if ($this->board->inBounds($nr,$nc) && !$this->board->isOccupied($nr,$nc)) {
                $moves[] = new Move($row,$col,$nr,$nc,[]);
            }
        }
        foreach ([-1,1] as $dr) foreach ([-1,1] as $dc) {
            $nr = $row+$dr; $nc = $col+$dc;
            if ($this->board->inBounds($nr,$nc)) {
                $target = $this->board->getPiece($nr,$nc);
                if ($target && $target->color !== $color) {
                    $jr = $nr+$dr; $jc = $nc+$dc;
                    if ($this->board->inBounds($jr,$jc) && !$this->board->isOccupied($jr,$jc)) {
                        $moves[] = new Move($row,$col,$jr,$jc,[[$nr,$nc]]);
                    }
                }
            }
        }
        return $moves;
    }

    private function getKingMoves(int $row, int $col, ColorType $color): array {
        $moves = [];
        foreach ([-1,1] as $dr) foreach ([-1,1] as $dc) {
            $r = $row+$dr; $c = $col+$dc;
            while ($this->board->inBounds($r,$c)) {
                if ($this->board->isOccupied($r,$c)) break;
                $moves[] = new Move($row,$col,$r,$c,[]);
                $r += $dr; $c += $dc;
            }
            $r = $row+$dr; $c = $col+$dc;
            while ($this->board->inBounds($r,$c)) {
                if ($this->board->isOccupied($r,$c)) {
                    $target = $this->board->getPiece($r,$c);
                    if ($target && $target->color !== $color) {
                        $br = $r+$dr; $bc = $c+$dc;
                        while ($this->board->inBounds($br,$bc)) {
                            if ($this->board->isOccupied($br,$bc)) break;
                            $moves[] = new Move($row,$col,$br,$bc,[[$r,$c]]);
                            break;
                        }
                    }
                    break;
                }
                $r += $dr; $c += $dc;
            }
        }
        return $moves;
    }

    public function getAllValidMoves(ColorType $color): array {
        $all = [];
        foreach ($this->board->findPieces($color) as [$r,$c]) {
            $p = $this->board->getPiece($r,$c);
            if (!$p) continue;
            if ($p->type === PieceType::MAN) $all = array_merge($all, $this->getManMoves($r,$c,$color));
            else $all = array_merge($all, $this->getKingMoves($r,$c,$color));
        }
        $caps = array_filter($all, fn($m) => count($m->captures) > 0);
        if (count($caps) > 0) return array_values($caps);
        return $all;
    }

    public function isValidMove(Move $move): bool {
        foreach ($this->getAllValidMoves($this->currentTurn) as $m) {
            if ($m->fromRow === $move->fromRow && $m->fromCol === $move->fromCol &&
                $m->toRow === $move->toRow && $m->toCol === $move->toCol) {
                if (count($m->captures) === count($move->captures)) {
                    $s1 = array_map(fn($c) => implode(',',$c), $m->captures);
                    $s2 = array_map(fn($c) => implode(',',$c), $move->captures);
                    sort($s1); sort($s2);
                    if ($s1 === $s2) return true;
                }
            }
        }
        return false;
    }

    public function makeMove(Move $move): bool {
        if (!$this->isValidMove($move)) return false;
        $p = $this->board->getPiece($move->fromRow, $move->fromCol);
        if (!$p) return false;
        $this->board->removePiece($move->fromRow, $move->fromCol);
        $this->board->placePiece($move->toRow, $move->toCol, $p);
        foreach ($move->captures as [$r,$c]) $this->board->removePiece($r,$c);
        if ($this->board->isKingRow($move->toRow, $p->color) && $p->type === PieceType::MAN) {
            $this->board->promoteToKing($move->toRow, $move->toCol);
        }
        $this->moveHistory[] = $move;
        $this->currentTurn = $this->opponent($this->currentTurn);
        return true;
    }

    public function isGameOver(): ?ColorType {
        if (count($this->board->findPieces(ColorType::WHITE)) === 0) return ColorType::BLACK;
        if (count($this->board->findPieces(ColorType::BLACK)) === 0) return ColorType::WHITE;
        if (count($this->getAllValidMoves($this->currentTurn)) === 0) return $this->opponent($this->currentTurn);
        return null;
    }

    public function parseMove(string $input): ?Move {
        $parts = preg_split('/\s+/', trim($input));
        if (count($parts) < 2) return null;
        $coords = [];
        foreach ($parts as $p) {
            if (!preg_match('/^([A-Ja-j])([1-9]|10)$/', $p, $m)) return null;
            $col = ord(strtoupper($m[1])) - ord('A');
            $row = intval($m[2]) - 1;
            $coords[] = [$row,$col];
        }
        if (count($coords) < 2) return null;
        [$fromR,$fromC] = $coords[0];
        [$toR,$toC] = $coords[count($coords)-1];
        $captures = [];
        for ($i=0; $i<count($coords)-1; $i++) {
            [$r1,$c1] = $coords[$i];
            [$r2,$c2] = $coords[$i+1];
            if (abs($r2-$r1) !== abs($c2-$c1)) return null;
            $dr = $r2 > $r1 ? 1 : -1;
            $dc = $c2 > $c1 ? 1 : -1;
            $cr = $r1 + $dr; $cc = $c1 + $dc;
            while ($cr !== $r2 || $cc !== $c2) {
                $piece = $this->board->getPiece($cr, $cc);
                if (!$piece || $piece->color === $this->currentTurn) return null;
                $captures[] = [$cr,$cc];
                $cr += $dr; $cc += $dc;
            }
        }
        return new Move($fromR,$fromC,$toR,$toC,$captures);
    }

    public function display(bool $useColor = true): void {
        echo "  ";
        for ($c=0; $c<Board::SIZE; $c++) echo chr(65+$c) . " ";
        echo "\n";
        for ($r=Board::SIZE-1; $r>=0; $r--) {
            printf("%2d ", $r+1);
            for ($c=0; $c<Board::SIZE; $c++) {
                $p = $this->board->getPiece($r,$c);
                $bg = ($r+$c)%2===0 ? 'white' : 'black';
                if (!$p) {
                    if ($useColor) echo colorize(" . ", RESET, $bg==='white' ? BG_WHITE : BG_BLACK);
                    else echo " . ";
                } else {
                    $sym = $p->color === ColorType::WHITE ? 'W' : 'B';
                    $sym .= $p->type === PieceType::KING ? '*' : ' ';
                    $fg = $bg==='white' ? BLACK : WHITE;
                    if ($useColor) echo colorize($sym, $fg, $bg==='white' ? BG_WHITE : BG_BLACK);
                    else echo $sym . " ";
                }
            }
            echo "\n";
        }
        echo "Ход: " . ($this->currentTurn === ColorType::WHITE ? "Белые" : "Чёрные") . "\n";
    }

    public function getAIMove(): ?Move {
        $moves = $this->getAllValidMoves($this->currentTurn);
        if (count($moves) === 0) return null;
        $caps = array_filter($moves, fn($m) => count($m->captures) > 0);
        if (count($caps) > 0) return $caps[array_rand($caps)];
        return $moves[array_rand($moves)];
    }

    public function saveState(string $filename): void {
        $data = [
            'board' => array_map(fn($r) => array_map(fn($c) => $c ? ['color'=>$c->color->name, 'type'=>$c->type->name] : null, $r), $this->board->grid),
            'turn' => $this->currentTurn->name,
            'history' => array_map(fn($m) => ['fromRow'=>$m->fromRow,'fromCol'=>$m->fromCol,'toRow'=>$m->toRow,'toCol'=>$m->

File: Checkers.cs
// Checkers.cs
// Версия на C# с использованием record, enum, LINQ, цветного вывода, сохранения/загрузки JSON

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Text.RegularExpressions;

namespace Checkers
{
    // ANSI
    public static class Ansi
    {
        public const string Reset = "\x1b[0m";
        public const string Black = "\x1b[30m";
        public const string Red = "\x1b[31m";
        public const string Green = "\x1b[32m";
        public const string Yellow = "\x1b[33m";
        public const string Blue = "\x1b[34m";
        public const string Magenta = "\x1b[35m";
        public const string Cyan = "\x1b[36m";
        public const string White = "\x1b[37m";
        public const string BgBlack = "\x1b[40m";
        public const string BgWhite = "\x1b[47m";

        public static string Colorize(string text, string fg = Reset, string bg = "")
        {
            return fg + bg + text + Reset;
        }
    }

    public enum ColorType { WHITE, BLACK }
    public enum PieceType { MAN, KING }

    public record Piece(ColorType Color, PieceType Type);

    public record Move(int FromRow, int FromCol, int ToRow, int ToCol, List<(int,int)> Captures);

    public class Board
    {
        public const int Size = 10;
        public Piece?[,] Grid = new Piece[Size, Size];

        public Board()
        {
            for (int r = 0; r < Size; r++)
                for (int c = 0; c < Size; c++)
                    if ((r + c) % 2 == 1)
                    {
                        if (r < 4) Grid[r, c] = new Piece(ColorType.WHITE, PieceType.MAN);
                        else if (r > 5) Grid[r, c] = new Piece(ColorType.BLACK, PieceType.MAN);
                    }
        }

        public bool InBounds(int r, int c) => r >= 0 && r < Size && c >= 0 && c < Size;
        public Piece? GetPiece(int r, int c) => InBounds(r, c) ? Grid[r, c] : null;
        public bool IsOccupied(int r, int c) => GetPiece(r, c) != null;
        public void PlacePiece(int r, int c, Piece p) => Grid[r, c] = p;
        public void RemovePiece(int r, int c) => Grid[r, c] = null;
        public List<(int,int)> FindPieces(ColorType color)
        {
            var res = new List<(int,int)>();
            for (int r = 0; r < Size; r++)
                for (int c = 0; c < Size; c++)
                    if (Grid[r, c]?.Color == color)
                        res.Add((r, c));
            return res;
        }
        public bool IsKingRow(int r, ColorType color) =>
            (color == ColorType.WHITE && r == Size - 1) || (color == ColorType.BLACK && r == 0);
        public void PromoteToKing(int r, int c)
        {
            var p = Grid[r, c];
            if (p != null && p.Type == PieceType.MAN)
                Grid[r, c] = new Piece(p.Color, PieceType.KING);
        }
        public Board Clone()
        {
            var b = new Board();
            for (int r = 0; r < Size; r++)
                for (int c = 0; c < Size; c++)
                    b.Grid[r, c] = Grid[r, c] != null ? new Piece(Grid[r, c].Color, Grid[r, c].Type) : null;
            return b;
        }
    }

    public class Game
    {
        public Board Board { get; private set; }
        public ColorType CurrentTurn { get; private set; }
        public bool TwoPlayers { get; }
        public List<Move> MoveHistory { get; } = new();

        public Game(bool twoPlayers)
        {
            Board = new Board();
            CurrentTurn = ColorType.WHITE;
            TwoPlayers = twoPlayers;
        }

        private ColorType Opponent(ColorType c) => c == ColorType.WHITE ? ColorType.BLACK : ColorType.WHITE;

        private IEnumerable<Move> GetManMoves(int row, int col, ColorType color)
        {
            var moves = new List<Move>();
            var dirs = color == ColorType.WHITE ? new (int,int)[]{(1,-1),(1,1)} : new (int,int)[]{(-1,-1),(-1,1)};
            foreach (var (dr,dc) in dirs)
            {
                int nr = row+dr, nc = col+dc;
                if (Board.InBounds(nr,nc) && !Board.IsOccupied(nr,nc))
                    moves.Add(new Move(row,col,nr,nc,new List<(int,int)>()));
            }
            foreach (int dr in new[]{-1,1})
                foreach (int dc in new[]{-1,1})
                {
                    int nr = row+dr, nc = col+dc;
                    if (Board.InBounds(nr,nc))
                    {
                        var target = Board.GetPiece(nr,nc);
                        if (target != null && target.Color != color)
                        {
                            int jr = nr+dr, jc = nc+dc;
                            if (Board.InBounds(jr,jc) && !Board.IsOccupied(jr,jc))
                                moves.Add(new Move(row,col,jr,jc,new List<(int,int)>{(nr,nc)}));
                        }
                    }
                }
            return moves;
        }

        private IEnumerable<Move> GetKingMoves(int row, int col, ColorType color)
        {
            var moves = new List<Move>();
            foreach (int dr in new[]{-1,1})
                foreach (int dc in new[]{-1,1})
                {
                    int r = row+dr, c = col+dc;
                    while (Board.InBounds(r,c))
                    {
                        if (Board.IsOccupied(r,c)) break;
                        moves.Add(new Move(row,col,r,c,new List<(int,int)>()));
                        r += dr; c += dc;
                    }
                    r = row+dr; c = col+dc;
                    while (Board.InBounds(r,c))
                    {
                        if (Board.IsOccupied(r,c))
                        {
                            var target = Board.GetPiece(r,c);
                            if (target != null && target.Color != color)
                            {
                                int br = r+dr, bc = c+dc;
                                while (Board.InBounds(br,bc))
                                {
                                    if (Board.IsOccupied(br,bc)) break;
                                    moves.Add(new Move(row,col,br,bc,new List<(int,int)>{(r,c)}));
                                    break;
                                }
                            }
                            break;
                        }
                        r += dr; c += dc;
                    }
                }
            return moves;
        }

        public List<Move> GetAllValidMoves(ColorType color)
        {
            var all = new List<Move>();
            foreach (var (r,c) in Board.FindPieces(color))
            {
                var p = Board.GetPiece(r,c);
                if (p == null) continue;
                if (p.Type == PieceType.MAN) all.AddRange(GetManMoves(r,c,color));
                else all.AddRange(GetKingMoves(r,c,color));
            }
            var caps = all.Where(m => m.Captures.Count > 0).ToList();
            if (caps.Count > 0) return caps;
            return all;
        }

        public bool IsValidMove(Move move)
        {
            foreach (var m in GetAllValidMoves(CurrentTurn))
            {
                if (m.FromRow == move.FromRow && m.FromCol == move.FromCol &&
                    m.ToRow == move.ToRow && m.ToCol == move.ToCol)
                {
                    if (m.Captures.Count == move.Captures.Count)
                    {
                        var s1 = new HashSet<string>(m.Captures.Select(t => $"{t.Item1},{t.Item2}"));
                        var s2 = new HashSet<string>(move.Captures.Select(t => $"{t.Item1},{t.Item2}"));
                        if (s1.SetEquals(s2)) return true;
                    }
                }
            }
            return false;
        }

        public bool MakeMove(Move move)
        {
            if (!IsValidMove(move)) return false;
            var p = Board.GetPiece(move.FromRow, move.FromCol);
            if (p == null) return false;
            Board.RemovePiece(move.FromRow, move.FromCol);
            Board.PlacePiece(move.ToRow, move.ToCol, p);
            foreach (var (r,c) in move.Captures)
                Board.RemovePiece(r,c);
            if (Board.IsKingRow(move.ToRow, p.Color) && p.Type == PieceType.MAN)
                Board.PromoteToKing(move.ToRow, move.ToCol);
            MoveHistory.Add(move);
            CurrentTurn = Opponent(CurrentTurn);
            return true;
        }

        public ColorType? IsGameOver()
        {
            if (Board.FindPieces(ColorType.WHITE).Count == 0) return ColorType.BLACK;
            if (Board.FindPieces(ColorType.BLACK).Count == 0) return ColorType.WHITE;
            if (GetAllValidMoves(CurrentTurn).Count == 0) return Opponent(CurrentTurn);
            return null;
        }

        public Move? ParseMove(string input)
        {
            var parts = input.Trim().Split(' ', StringSplitOptions.RemoveEmptyEntries);
            if (parts.Length < 2) return null;
            var coords = new List<(int,int)>();
            var re = new Regex(@"^([A-Ja-j])([1-9]|10)$");
            foreach (var p in parts)
            {
                var m = re.Match(p);
                if (!m.Success) return null;
                int col = char.ToUpper(m.Groups[1].Value[0]) - 'A';
                int row = int.Parse(m.Groups[2].Value) - 1;
                coords.Add((row, col));
            }
            if (coords.Count < 2) return null;
            var from = coords[0];
            var to = coords[^1];
            var captures = new List<(int,int)>();
            for (int i=0; i<coords.Count-1; i++)
            {
                var (r1,c1) = coords[i];
                var (r2,c2) = coords[i+1];
                if (Math.Abs(r2-r1) != Math.Abs(c2-c1)) return null;
                int dr = r2>r1 ? 1 : -1;
                int dc = c2>c1 ? 1 : -1;
                int cr = r1+dr, cc = c1+dc;
                while (cr != r2 || cc != c2)
                {
                    var piece = Board.GetPiece(cr, cc);
                    if (piece == null || piece.Color == CurrentTurn) return null;
                    captures.Add((cr, cc));
                    cr += dr; cc += dc;
                }
            }
            return new Move(from.Item1, from.Item2, to.Item1, to.Item2, captures);
        }

        public void Display(bool useColor)
        {
            Console.Write("  ");
            for (int c=0; c<Board.Size; c++) Console.Write($"{(char)('A'+c)} ");
            Console.WriteLine();
            for (int r=Board.Size-1; r>=0; r--)
            {
                Console.Write($"{r+1,2} ");
                for (int c=0; c<Board.Size; c++)
                {
                    var p = Board.GetPiece(r,c);
                    string bg = (r+c)%2==0 ? "white" : "black";
                    if (p == null)
                    {
                        if (useColor) Console.Write(Ansi.Colorize(" . ", Ansi.Reset, bg=="white" ? Ansi.BgWhite : Ansi.BgBlack));
                        else Console.Write(" . ");
                    }
                    else
                    {
                        string sym = p.Color == ColorType.WHITE ? "W" : "B";
                        sym += p.Type == PieceType.KING ? "*" : " ";
                        string fg = bg=="white" ? Ansi.Black : Ansi.White;
                        if (useColor) Console.Write(Ansi.Colorize(sym, fg, bg=="white" ? Ansi.BgWhite : Ansi.BgBlack));
                        else Console.Write(sym + " ");
                    }
                }
                Console.WriteLine();
            }
            Console.WriteLine($"Ход: {(CurrentTurn == ColorType.WHITE ? "Белые" : "Чёрные")}");
        }

        public Move? GetAIMove()
        {
            var moves = GetAllValidMoves(CurrentTurn);
            if (moves.Count == 0) return null;
            var caps = moves.Where(m => m.Captures.Count > 0).ToList();
            if (caps.Count > 0) return caps[new Random().Next(caps.Count)];
            return moves[new Random().Next(moves.Count)];
        }

        public void SaveState(string filename)
        {
            var state = new
            {
                board = Enumerable.Range(0, Board.Size).Select(r =>
                    Enumerable.Range(0, Board.Size).Select(c =>
                        Board.Grid[r,c] != null ? new { color = Board.Grid[r,c].Color.ToString(), type = Board.Grid[r,c].Type.ToString() } : null
                    ).ToArray()
                ).ToArray(),
                turn = CurrentTurn.ToString(),
                history = MoveHistory.Select(m => new { m.FromRow, m.FromCol, m.ToRow, m.ToCol, Captures = m.Captures.Select(t => new { t.Item1, t.Item2 }).ToArray() }).ToArray()
            };
            var json = JsonSerializer.Serialize(state, new JsonSerializerOptions { WriteIndented = true });
            File.WriteAllText(filename, json);
        }

        public void LoadState(string filename)
        {
            var json = File.ReadAllText(filename);
            var state = JsonSerializer.Deserialize<JsonElement>(json);
            var boardData = state.GetProperty("board").EnumerateArray().Select(row => row.EnumerateArray().Select(cell =>
                cell.ValueKind != JsonValueKind.Null ? new Piece(
                    Enum.Parse<ColorType>(cell.GetProperty("color").GetString()),
                    Enum.Parse<PieceType>(cell.GetProperty("type").GetString())
                ) : null
            ).ToArray()).ToArray();
            var b = new Board();
            for (int r=0; r<Board.Size; r++)
                for (int c=0; c<Board.Size; c++)
                    b.Grid[r,c] = boardData[r][c];
            Board = b;
            CurrentTurn = Enum.Parse<ColorType>(state.GetProperty("turn").GetString());
            // history не восстанавливаем
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            bool useColor = true;
            bool twoPlayers = false;
            string saveFile = null, loadFile = null;
            for (int i=0; i<args.Length; i++)
            {
                if (args[i] == "--no-color") useColor = false;
                else if (args[i] == "--color") useColor = true;
                else if (args[i] == "--two-players") twoPlayers = true;
                else if (args[i] == "--save" && i+1 < args.Length) saveFile = args[++i];
                else if (args[i] == "--load" && i+1 < args.Length) loadFile = args[++i];
                else if (args[i] == "--help")
                {
                    Console.WriteLine("Использование: dotnet run -- [опции]");
                    Console.WriteLine("  --color               Цветной вывод (по умолчанию)");
                    Console.WriteLine("  --no-color            Отключить цвет");
                    Console.WriteLine("  --two-players         Режим для двух игроков");
                    Console.WriteLine("  --save <file>         Сохранять состояние");
                    Console.WriteLine("  --load <file>         Загрузить состояние");
                    return;
                }
            }

            var game = new Game(twoPlayers);
            if (loadFile != null) game.LoadState(loadFile);

            while (true)
            {
                game.Display(useColor);
                var winner = game.IsGameOver();
                if (winner != null)
                {
                    Console.WriteLine($"Победили {(winner == ColorType.WHITE ? "Белые" : "Чёрные")}!");
                    break;
                }
                if (game.TwoPlayers)
                {
                    string name = game.CurrentTurn == ColorType.WHITE ? "Белые" : "Чёрные";
                    Console.Write($"Ход {name}. Введите ход: ");
                    string input = Console.ReadLine();
                    if (input == "quit" || input == "exit") break;
                    var move = game.ParseMove(input);
                    if (move == null) { Console.WriteLine("Неверный формат хода."); continue; }
                    if (game.MakeMove(move))
                    {
                        Console.WriteLine("Ход принят.");
                        if (saveFile != null) game.SaveState(saveFile);
                    }
                    else Console.WriteLine("Недопустимый ход.");
                }
                else
                {
                    if (game.CurrentTurn == ColorType.BLACK)
                    {
                        var move = game.GetAIMove();
                        if (move == null) { Console.WriteLine("Компьютер не может ходить."); break; }
                        game.MakeMove(move);
                        Console.WriteLine($"Компьютер сходил: {(char)('A'+move.FromCol)}{move.FromRow+1} -> {(char)('A'+move.ToCol)}{move.ToRow+1}");
                        if (saveFile != null) game.SaveState(saveFile);
                    }
                    else
                    {
                        Console.Write("Ваш ход (например, a1 b2): ");
                        string input = Console.ReadLine();
                        if (input == "quit" || input == "exit") break;
                        var move = game.ParseMove(input);
                        if (move == null) { Console.WriteLine("Неверный формат хода."); continue; }
                        if (game.MakeMove(move))
                        {
                            Console.WriteLine("Ход принят.");
                            if (saveFile != null) game.SaveState(saveFile);
                        }
                        else Console.WriteLine("Недопустимый ход.");
                    }
                }
            }
        }
    }
}
